ragna/source_storages/_qdrant.py

```diff
@@ -0,0 +1,249 @@
+                ),
+            )
+        elif no_corpuses:
+            raise_no_corpuses_available(self)
+        elif non_existing_corpus:
+            raise_non_existing_corpus(self, corpus_name)
+
+    def list_metadata(
+        self, corpus_name: Optional[str] = None
+    ) -> dict[str, dict[str, tuple[str, list[Any]]]]:
+        if corpus_name is None:
+            corpus_names = self.list_corpuses()
+        else:
+            corpus_names = [corpus_name]
+
+        metadata = {}
+        for corpus_name in corpus_names:
+            points, _offset = self._client.scroll(
+                collection_name=corpus_name, with_payload=True
+            )
+
+            corpus_metadata = defaultdict(set)
+            for point in points:
+                for key, value in point.payload.items():
+                    if any(
+                        [
+                            (key.startswith("__") and key.endswith("__")),
+                            key == self.DOC_CONTENT_KEY,
+                            not value,
+                        ]
+                    ):
+                        continue
+
+                    corpus_metadata[key].add(value)
+
+            metadata[corpus_name] = {
+                key: ({type(value).__name__ for value in values}.pop(), sorted(values))
+                for key, values in corpus_metadata.items()
+            }
+
+        return metadata
+
+    def store(
+        self,
+        corpus_name: str,
+        documents: list[Document],
+        *,
+        chunk_size: int = 500,
+        chunk_overlap: int = 250,
+    ) -> None:
+        from qdrant_client import models
+
+        self._ensure_table(corpus_name, create=True)
+
+        points = []
+        for document in documents:
+            for chunk in self._chunk_pages(
+                document.extract_pages(),
+                chunk_size=chunk_size,
+                chunk_overlap=chunk_overlap,
+            ):
+                points.append(
+                    models.PointStruct(
+                        id=str(uuid.uuid4()),
+                        vector=self._embedding_function([chunk.text])[0],
+                        payload={
+                            "document_id": str(document.id),
+                            "document_name": document.name,
+                            **document.metadata,
+                            "__page_numbers__": self._page_numbers_to_str(
+                                chunk.page_numbers
+                            ),
+                            "__num_tokens__": chunk.num_tokens,
+                            self.DOC_CONTENT_KEY: chunk.text,
+                        },
+                    )
+                )
+
+        self._client.upsert(collection_name=corpus_name, points=points)
+
+    def _build_condition(self, operator, key, value):
+        from qdrant_client import models
+
+        if operator == MetadataOperator.EQ:
```

Comment:
Please add a link to the documentation as comment here
```suggestion
        # See https://qdrant.tech/documentation/concepts/filtering/#range
        if operator == MetadataOperator.EQ:
```